### PR TITLE
Balder/make thread safe storage policy

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@jobergum and @ollivir FYI

**Review by baldersheim — COMMENTED** [review]
Now I think there are less possibilities for race condition.

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/SlobrokPolicy.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (vekterli) [inline documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/SlobrokPolicy.java]: Consider updating copyright since this is a new file

**Review by vekterli — APPROVED** [review]
:+1: 🎉

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/AsyncInitializationPolicy.java [hunk 2]
```diff
@@ -58,7 +58,7 @@
         initState = InitState.NOT_STARTED;
     }
 
-    void needAsynchronousInitialization() {
+    synchronized void needAsynchronousInitialization() {
```
[the remark this answers is not in the document]
Comment (vekterli) [reply documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/AsyncInitializationPolicy.java]: Outdated pre-async remove comment which snuck in, ignore.

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/SlobrokPolicy.java (hunk 1 above)
Comment (baldersheim) [reply documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/SlobrokPolicy.java]: Fixed

**Pull request merged**